src/Template/Element/index/bulk_actions/form_end.ctp

```diff
@@ -0,0 +1,30 @@
+<?php
+if (empty($bulkActions)) {
+    return;
+}
+
+$this->Form->templates([
+    'submitContainer' => '{{content}}',
+]);
+
+$submitButton = $this->Form->submit(__d('crud', 'Apply'), [
+    'class' => 'btn btn-success btn-bulk-apply',
+    'div' => false,
+    'name' => '_bulk',
+]);
+$this->Form->templates([
+    'inputContainer' => '<div class="form-group bulk-action-submit {{required}}">{{content}}{{help}}</div>',
+    'select' => '<div class="col-sm-10"><select name="{{name}}"{{attrs}}>{{content}}</select>' . $submitButton . '</div>',
```

Comment:
I don't really understand the reasoning there.

Calling `Form->templates` will override the config for those keys, leaving the Form helper in a different state. If the goal is to change templates only for this one element, the form helper should be in the same state after the element as before - which it won't be. There shouldn't be any config in template files though.

I might be commenting on how the form helper works rather than this though - isn't it possible to switch template sets at run time, without re-creating the form helper object?